### PR TITLE
MBS-11293 Use test retry Gradle plugin for unit tests

### PR DESCRIPTION
**Pull request by eugene-krivobokov**
https://github.com/gradle/test-retry-gradle-plugin

Now we can retry unstable tests (unit, gradle, integration).

### CODE REVIEW

File: build-logic/testing-convention/src/main/kotlin/convention.unit-testing.gradle.kts [hunk 1]
```diff
@@ -35,6 +40,14 @@ tasks.withType<Test>().configureEach {
         "isInvokedFromIde",
         gradle.startParameter.allInitScripts.find { it.name.contains("ijtestinit") } != null
     )
+    if (isCi.getOrElse(false)) {
+        retry {
+            maxRetries.set(1)
+            filter {
+                includeAnnotationClasses.add("com.avito.test.Flaky")
```
Comment (dsvoronin) [inline build-logic/testing-convention/src/main/kotlin/convention.unit-testing.gradle.kts:47]: i like this API so much!
we should reuse it for our ui tests

File: subprojects/common/junit-utils/src/main/java/com/avito/test/Flaky.kt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package com.avito.test
+
+/**
+ * Unstable test marker.
+ * It's used for retry logic.
+ */
+annotation class Flaky(
```
Comment (dsvoronin) [inline subprojects/common/junit-utils/src/main/java/com/avito/test/Flaky.kt:7]: I think you should specify retention

**Review by dsvoronin — APPROVED** [review]

**Review by sergioboi — APPROVED** [review]

File: subprojects/common/junit-utils/src/main/java/com/avito/test/Flaky.kt (hunk 2 above)
Comment (eugene-krivobokov) [reply subprojects/common/junit-utils/src/main/java/com/avito/test/Flaky.kt:7]: Ok, I'll check it out in next PR

Comment (sergioboi) [reply subprojects/common/junit-utils/src/main/java/com/avito/test/Flaky.kt:7]: I don't understand what the retention meaning is in that context

Comment (eugene-krivobokov) [reply subprojects/common/junit-utils/src/main/java/com/avito/test/Flaky.kt:7]: `kotlin.annotation.AnnotationRetention`
As I think, the idea is to use the minimum required retention, to store only needed information.

**Pull request merged**